docs/design/gateway/lwM2M-gateway.md

```diff
@@ -0,0 +1,176 @@
+# LwM2M Gateway design
+
+## Why need LwM2M Gateway
+
+For telemetryService only support push data to the server, but for LwM2M protocol, it support both push and pull data from the device. it is hard to implement the pull data feature in the telemetryService.
+
+So we need a gateway make deviceShifu to adapt the LwM2M protocol. to support pull data call from server and auto push data to the server.
+
+## Goal
+
+### Design Goal
+
+- [LwM2M protocol using v1.0.x version](https://www.openmobilealliance.org/release/LightweightM2M/V1_0-20170208-A/OMA-TS-LightweightM2M-V1_0-20170208-A.pdf).
+- LwM2M protocol under UDP.
+- Datagram Transport Layer Security (DTLS) support.
+- Support using LwM2M protocol to communicate with the server.
+- Support both `read` and `write` requests.
+- Support Notify and Observe feature.
+
+### Non-Goal
+
+- Support LwM2M v1.1.x or later version.
+- Over TCP or other protocol.
+- Bootstrap Server.
+- Support all the LwM2M Object.
+- Support all the LwM2M Resource.
+
+## LwM2M Gateway Design
+
+For LwM2M Gateway, it will as a LwM2M client to connect a server and it will handle all request from the server over the LwM2M protocol. 
+
+When a device enable the gateway feature, it will register to the gateway and the gateway will call the server to update the device info. Each device will have a unique ObjectId like `/33953` and their.
+instruction will be a instance of the ObjectId like `/33953/1`.
+
+```mermaid
+flowchart BT
+
+ls[lwm2m-server]
+
+subgraph EdgeNode
+    subgraph Shifu
+      subgraph ds1[deviceshifu-lwM2M]
+          dsh[deviceshifu-http]
+          gl1[lwm2m-gateway]
+          dsh <-->|HTTP| gl1
+      end
+      subgraph ds2[deviceshifu-MQTT]
+          dsm[deviceshifu-MQTT]
+          gl2[lwm2m-gateway]
+          dsm <-->|HTTP| gl2
+      end
+    end
+end
+
+
+dh[device-http]
+dm[device-mqtt]
+
+
+dh -->|HTTP| dsh
+gl1 -->|lwM2M| ls
+
+dm -->|MQTT| dsm
+gl2 -->|lwM2M| ls 
+```
+
+### What will the gateway do?
```

Comment:
make a clear statement about "server", it seems the reader would confuse about this concept